src/lib/core/components/visualizations/implementations/HistogramVisualization.tsx

```diff
@@ -878,15 +672,38 @@ function HistogramPlotWithControls({
     setTruncatedDependentAxisWarning,
   ]);
 
-  // send histogramProps with additional props
-  const histogramPlotProps = {
-    ...histogramProps,
-    // axis range control
+  const plotRef = useUpdateThumbnailEffect(
+    updateThumbnail,
+    plotContainerStyles,
+    [
+      data,
+      checkedLegendItems,
+      vizConfig.dependentAxisLogScale,
+      vizConfig.dependentAxisRange,
+    ]
+  );
+
+  const overlayLabel = variableDisplayWithUnit(overlayVariable);
+
+  const histogramProps: HistogramProps = {
```

Comment:
@dmfalke I am testing this grand merge (😄) for my past commits like axis label power format and proportion bugs fir histogram and barplot Vizs. They seem to work fine for me. However, I found a minor bug on histogram Viz, which is probably inherited from Bob's Viz separation works. There is a missing prop in this histogramProps, so the following should also be added - histogram plot style is not being applied without it, e.g., no box shadow, height, etc.

`containerStyles: !isFaceted(data.value) ? plotContainerStyles : undefined,`

Reply:
Thanks @moontrip! I will add this.

Reply:
Actually, I think this is addressed here:
https://github.com/VEuPathDB/web-eda/blob/289293be9c1f6a5a738de6e258fa0e0f8559854e/src/lib/core/components/visualizations/implementations/HistogramVisualization.tsx#L719-L741

Reply:
Oh, ignore that. That is something different.